stripe/src/main/java/com/stripe/android/model/wallets/MasterpassWallet.java

```diff
@@ -0,0 +1,126 @@
+package com.stripe.android.model.wallets;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.stripe.android.utils.ObjectUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.stripe.android.model.StripeJsonUtils.optString;
+
+public final class MasterpassWallet extends Wallet {
+    private static final String FIELD_BILLING_ADDRESS = "billing_address";
+    private static final String FIELD_EMAIL = "email";
+    private static final String FIELD_NAME = "name";
+    private static final String FIELD_SHIPPING_ADDRESS = "shipping_address";
+
+    @Nullable public final Address billingAddress;
+    @Nullable public final String email;
+    @Nullable public final String name;
+    @Nullable public final Address shippingAddress;
+
+    private MasterpassWallet(@NonNull Builder builder) {
+        super(Type.Masterpass, builder);
+        billingAddress = builder.mBillingAddress;
+        email = builder.mEmail;
+        name = builder.mName;
+        shippingAddress = builder.mShippingAddress;
+    }
+
+    @NonNull
+    @Override
+    Map<String, Object> getWalletTypeMap() {
+        final AbstractMap<String, Object> wallet = new HashMap<>();
+        wallet.put(FIELD_BILLING_ADDRESS,
+                billingAddress != null ? billingAddress.toMap() : null);
+        wallet.put(FIELD_EMAIL, email);
+        wallet.put(FIELD_NAME, name);
+        wallet.put(FIELD_SHIPPING_ADDRESS,
+                shippingAddress != null ? shippingAddress.toMap() : null);
+        return wallet;
+    }
+
+    @NonNull
+    @Override
+    JSONObject getWalletTypeJson() {
+        final JSONObject wallet = new JSONObject();
+        try {
```

Comment:
ooc why this this try/catch necessary?

Reply:
`JSONObject.put` throws a `JSONException` (e.g. if the key is null) so it needs to be explicitly caught